tests/integration/producer/test_transactions.py

```diff
@@ -0,0 +1,189 @@
+
+    })
+
+    producer.init_transactions()
+    transactional_produce(producer, output_topic, 100)
+    producer.commit_transaction()
+
+    assert consume_committed(kafka_cluster.client_conf(), output_topic) == 100
+
+
+def test_abort_transaction(kafka_cluster):
+    output_topic = kafka_cluster.create_topic("output_topic" + str(uuid1()))
+
+    producer = kafka_cluster.producer({
+        'transactional.id': 'example_transactional_id',
+        'error_cb': my_error_cb,
+    })
+
+    producer.init_transactions()
+    transactional_produce(producer, output_topic, 100)
+    producer.abort_transaction()
+
+    assert consume_committed(kafka_cluster.client_conf(), output_topic) == 0
+
+
+def test_abort_retry_commit_transaction(kafka_cluster):
+    output_topic = kafka_cluster.create_topic("output_topic" + str(uuid1()))
+
+    producer = kafka_cluster.producer({
+        'transactional.id': 'example_transactional_id',
+        'error_cb': my_error_cb,
+    })
+
+    producer.init_transactions()
+    transactional_produce(producer, output_topic, 100)
+    producer.abort_transaction()
+
+    transactional_produce(producer, output_topic, 25)
+    producer.commit_transaction()
+
+    assert consume_committed(kafka_cluster.client_conf(), output_topic) == 25
+
+
+def test_send_offsets_committed_transaction(kafka_cluster):
+    input_topic = kafka_cluster.create_topic("input_topic" + str(uuid1()))
+    output_topic = kafka_cluster.create_topic("output_topic" + str(uuid1()))
+
+    producer = kafka_cluster.producer({
+        'client.id': 'producer1',
+        'transactional.id': 'example_transactional_id',
+        'error_cb': my_error_cb,
+    })
+
+    consumer_conf = {
+        'group.id': str(uuid1()),
+        'auto.offset.reset': 'earliest',
+        'isolation.level': 'read_committed',
+        'enable.auto.commit': False,
+        'enable.partition.eof': True,
+        'error_cb': my_error_cb
+    }
+    consumer_conf.update(kafka_cluster.client_conf())
+    consumer = Consumer(consumer_conf)
+
+    kafka_cluster.seed_topic(input_topic)
+    consumer.subscribe([input_topic])
+
+    read_all_msgs(consumer)
+
+    producer.init_transactions()
+    transactional_produce(producer, output_topic, 100)
+
+    print("=== Sending offsets {} to transaction ===".format(consumer.position(consumer.assignment())))
+    producer.send_offsets_to_transaction(consumer_conf.get('group.id'),
+                                         consumer.position(consumer.assignment()))
+    producer.commit_transaction()
+
+    producer = kafka_cluster.producer({
+        'client.id': 'producer2',
+        'transactional.id': 'example_transactional_id',
+        'error_cb': my_error_cb
+    })
+
+    # ensure offset commits are visible prior to sending FetchOffsets request
+    producer.init_transactions()
+
+    committed = consumer.committed(consumer.assignment())
+    print("=== Committed offsets for {} ===".format(committed))
+    committed = [tp.offset for tp in committed]
```

Comment:
changing types of an existing variable is confusing to the reader, perhaps committed_offsets ?